File: server/app/api/app/memory.py
from __future__ import annotations

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.auth import get_current_user
from app.core.database import get_db
from app.models.user import User
from app.schemas.app_memory import MemoryGreetingPayload, MemorySuggestPayload
from app.services.memory_service import MemoryService

router = APIRouter(tags=["app-memory"])


@router.get("/memory/greeting", response_model=MemoryGreetingPayload)
async def get_memory_greeting(
    user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
) -> MemoryGreetingPayload:
    return MemoryGreetingPayload(**(await MemoryService(db).get_greeting(user=user)))


@router.get("/memory/suggest", response_model=MemorySuggestPayload)
async def get_memory_suggest(
    user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
) -> MemorySuggestPayload:
    return MemorySuggestPayload(**(await MemoryService(db).get_suggestions(user=user)))

File: apps/xince-app/src/shared/services/memory.ts
import type { MemoryGreetingPayload, MemorySuggestPayload } from "@/shared/models/memory";
import { request } from "@/shared/services/http";

export function fetchMemoryGreeting() {
  return request<MemoryGreetingPayload>("/memory/greeting");
}

export function fetchMemorySuggest() {
  return request<MemorySuggestPayload>("/memory/suggest");
}
